### PR TITLE
sys-firmware/edk2-ovmf: add 202405

### PR DESCRIPTION
**Pull request by takaki-s**
The current version of edk2-ovmf is pretty old and would like to have the latest version.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: sys-firmware/edk2-ovmf/edk2-ovmf-202405.ebuild [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+BUNDLED_OPENSSL_SUBMODULE_SHA="de90e54bbe82e5be4fb9608b6f5c308bb837d355"
+BUNDLED_BROTLI_SUBMODULE_SHA="f4153a09f87cbb9c826d8fc12c74642bb2d879ea"
+BUNDLED_MIPI_SYS_T_SUBMODULE_SHA="370b5944c046bab043dd8b133727b2135af7747a"
+BUNDLED_MBEDTLS_SUBMODULE_SHA="8c89224991adff88d53cd380f42a2baa36f91454"
+BUNDLED_LIBSPDM_SUBMODULE_SHA="828ef62524bcaeca4e90d0c021221e714872e2b5"
+
+# TODO: talk with tamiko about unbundling (mva)
+
+# TODO: the binary 202105 package currently lacks the preseeded
+#       OVMF_VARS.secboot.fd file (that we typically get from fedora)
```
Comment (juippis) [inline sys-firmware/edk2-ovmf/edk2-ovmf-202405.ebuild:23]: Are this still "TODO"? :)

File: sys-firmware/edk2-ovmf/edk2-ovmf-202405.ebuild [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	export EXTRA_LDFLAGS="-z notext"
+	export DLINK_FLAGS="-z notext"
+
+	emake ARCH=${TARGET_ARCH} -C BaseTools
+
+	. ./edksetup.sh
+
+	# Build all EFI firmware blobs:
+
+	mkdir -p ovmf
```
Comment (juippis) [inline sys-firmware/edk2-ovmf/edk2-ovmf-202405.ebuild:127]:
```suggestion
	mkdir -p ovmf || die
```

File: sys-firmware/edk2-ovmf/edk2-ovmf-202405.ebuild [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+	./OvmfPkg/build.sh \
+		-a "${TARGET_ARCH}" -b "${TARGET_NAME}" -t "${TARGET_TOOLS}" \
+		${BUILD_FLAGS} || die "OvmfPkg/build.sh failed"
+
+	cp Build/OvmfX64/*/FV/OVMF_*.fd ovmf/
+	rm -rf Build/OvmfX64
```
Comment (juippis) [inline sys-firmware/edk2-ovmf/edk2-ovmf-202405.ebuild:134]: We don't use `-f` for predictable files.
```suggestion
	rm -r Build/OvmfX64 || die
```

File: sys-firmware/edk2-ovmf/edk2-ovmf-202405.ebuild [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+	cp -rl "${WORKDIR}/public-mipi-sys-t-${BUNDLED_MIPI_SYS_T_SUBMODULE_SHA}"/* "MdePkg/Library/MipiSysTLib/mipisyst/" \
+		|| die "copying mipi-sys-t failed"
+	cp -rl "${WORKDIR}/mbedtls-${BUNDLED_MBEDTLS_SUBMODULE_SHA}"/* "CryptoPkg/Library/MbedTlsLib/mbedtls/" \
+		|| die "copying mbedtls failed"
+	cp -rl "${WORKDIR}/libspdm-${BUNDLED_LIBSPDM_SUBMODULE_SHA}"/* "SecurityPkg/DeviceSecurity/SpdmLib/libspdm" \
+		|| die "copying libspdm failed"
+
+	sed -i -r \
+		-e "/function SetupPython3/,/\}/{s,\\\$\(whereis python3\),${EPYTHON},g}" \
+		"${S}"/edksetup.sh || die "Fixing for correct Python3 support failed"
```
Comment (juippis) [inline sys-firmware/edk2-ovmf/edk2-ovmf-202405.ebuild:95]: This doesn't do anything.
```
SED: the following did not cause any changes
    sed -i -r -e "/function SetupPython3/,/\}/{s,\\\$\(whereis python3\),${EPYTHON},g}" "${S}"/edksetup.sh || die "Fixing for correct Python3 support failed";
no-op: -e /function SetupPython3/,/\}/{s,\$\(whereis python3\),python3.12,g}
```

**Review by juippis — APPROVED** [review]

File: sys-firmware/edk2-ovmf/edk2-ovmf-202405.ebuild (hunk 4 above)
Comment (takaki-s) [reply sys-firmware/edk2-ovmf/edk2-ovmf-202405.ebuild:95]: Thanks for checking! Seems like the SetupPython3 script got simplified a lot and no need to patch this anymore.

**Comment by henning-schild** [comment]
@ajakk maybe also bump sys-firmware/edk2-ovmf-bin

**Pull request closed without merging**